Add benchmark for TRSM

CodeplayBen opened the pull request:
Adds a benchmark for TRSM along with support for reference blas TRSM used for verification.

Bensuo committed:
Add benchmark for TRSM

ProGTX commented on include/utils/system_reference_blas.hpp:
```diff
@@ -257,6 +271,14 @@ void gemm(const char *transA, const char *transB, int m, int n, int k,
 }
 
+template <typename scalar_t>
+void trsm(char side, char triangle, char transpose, char diagonal, int m, int n,
```
I don't think this is part of the intended MR, I suppose this just needs a rebase on top of latest branch changes?

mcleary replied:
I will update this with Ben's commit. It needs some changes before merging

mcleary committed:
Add TRSM as a BLAS Level 3 Routine (#265)

* Add TRSM as a BLAS Level 3 Routine

- New function _trsm exposed to the user that can be used to solve Triangular Systems
- New kernel to invert the diagonal blocks of a matrix DiagonalBlocksInverter
- New utility function to fill a sycl-blas BufferIterator with a value
- Simple test that checks the result of the TRSM for a simple 4x4 system

* Add missing barrier for upper triangular case in diagonal blocks inverter

Add fully qualified sycl namespace

* Fix diagonal blocks inversion when matrix is upper triangular

* Add blas3_interface.h to the blas3 default_cpu backend

* Add unittests for trsm

* Reestructuring of the blas3 internals

Co-authored-by: Thales Sabino <thales@codeplay.com>
Co-authored-by: Mehdi Goli <mehdi.goli@codeplay.com>
Co-authored-by: Ben Tracy <50988088+CodeplayBen@users.noreply.github.com>
Co-authored-by: Peter Žužek <peterzuzek@gmail.com>
